### PR TITLE
Expose `RepoCard` at top level + few qol improvements

### PR DESCRIPTION
**Pull request by Wauplin**
Resolve https://github.com/huggingface/huggingface_hub/issues/1348.
No need to add a `SpaceCard` class, it wouldn't have any additional features.

I took the opportunity to add a few QOL improvement to the `CardData` object (`.get`, `__getitem__`, `__setitem__`, `__contains__`, `.pop`). More of these could be added but honestly not worth the hassle (those 5 cover 99% of the use cases). Could have been simpler to just inherit from `dict` but this way we can explicitly add an "export" step when serializing the metadata. 

Added some tests and completed a docstring.

### CODE REVIEW

File: src/huggingface_hub/repocard.py [hunk 1]
```diff
@@ -463,6 +463,17 @@ def from_template(  # type: ignore # violates Liskov property but easier to use
         return super().from_template(card_data, template_path, **template_kwargs)
 
 
+class SpaceCard:
+    """Space card is an alias for [`RepoCard`].
+
+    At the moment, it does not implement any specific logic. `SpaceCard` is defined for
```
Comment (merveenoyan) [inline src/huggingface_hub/repocard.py:469]: Do you think it makes sense to have Space-specific attributes in `RepoCard` under this instead? (e.g. like the one that tells us about duplicate)

**Review by merveenoyan — APPROVED** [review]
looks good, I just skimmed through it :')

File: src/huggingface_hub/repocard.py (hunk 1 above)
Comment (julien-c) [reply src/huggingface_hub/repocard.py:469]: yes!!

Comment (Wauplin) [reply src/huggingface_hub/repocard.py:469]: Defintely. Not useless in the end :smile:

Comment (Wauplin) [reply src/huggingface_hub/repocard.py]: @merveenoyan @julien-c added it in https://github.com/huggingface/huggingface_hub/pull/1354/commits/d5e71bb62e000acd2d4a63ad2067eba2b8526b4d

File: src/huggingface_hub/repocard.py [hunk 2]
```diff
@@ -787,18 +794,13 @@ def metadata_update(
                         card.data.eval_results.append(new_result)
         else:
             # Any metadata that is not a result metric
-            if (
-                hasattr(card.data, key)
-                and getattr(card.data, key) is not None
-                and not overwrite
-                and getattr(card.data, key) != value
-            ):
+            if card.data.get(key) is not None and not overwrite and card.data.get(key) != value:
```
Comment (nateraw) [inline src/huggingface_hub/repocard.py:797]: 🙏 nice

File: src/huggingface_hub/repocard_data.py [hunk 3]
```diff
@@ -362,6 +391,83 @@ def _to_dict(self, data_dict):
...
+        >>> card_data.to_dict()
+        {'title': 'Dreambooth Training', 'sdk': 'gradio', 'license': 'mit', 'duplicated_from': 'multimodalart/dreambooth-training'}
+        ```
+    """
+
+    def __init__(
+        self,
+        *,
+        title: Optional[str] = None,
+        sdk: Optional[str] = None,
```
Comment (nateraw) [inline src/huggingface_hub/repocard_data.py:445]: Should we fail if this is None after init? Fine either way...makes sense this dict-like wouldn't do that.

**Review by nateraw — APPROVED** [review]
LGTM! Thanks ❤️ this is super useful.

File: src/huggingface_hub/repocard_data.py (hunk 3 above)
Comment (Wauplin) [reply src/huggingface_hub/repocard_data.py:445]: I thought about it but I'd rather not. As I see it, `RepoCard` is here to help dealing with READMEs but does not really do type checking/data validation. I would rather keep it permissive for now and let the Hub do the validation job (if any).

**Comment by Wauplin** [comment]
Thanks for the reviews and feedback @merveenoyan @nateraw 
I'm merging this PR :)

**Pull request merged**